packages/amplication-client/src/Resource/create-resource/wizard-pages/CreateServiceAuth.tsx

```diff
@@ -1,13 +1,52 @@
-import React, { useCallback } from "react";
+import React, { useCallback, useEffect, useMemo } from "react";
+import { PluginLogo } from "../../../Plugins/PluginLogo";
+import usePlugins from "../../../Plugins/hooks/usePlugins";
+import { EnumCodeGenerator } from "../../../models";
 import "../CreateServiceWizard.scss";
 import { CreateServiceWizardLayout as Layout } from "../CreateServiceWizardLayout";
 import { LabelDescriptionSelector } from "./LabelDescriptionSelector";
 import { WizardStepProps } from "./interfaces";
-import authModuleImage from "../../../assets/images/auth-module.svg";
-import ImgSvg from "./ImgSvg";
+
+const NODE_AUTH_PLUGINS = ["auth-jwt", "auth-basic", "auth-keycloak"];
+const DOTNET_AUTH_PLUGINS = ["dotnet-auth-jwt"];
+
+const OVERRIDE_PLUGIN_DESCRIPTION = {
+  "auth-jwt": "Use JSON Web Token (JWT) authentication",
+  "auth-basic": "Use basic (username-password) authentication",
+  "auth-keycloak": "Use Keycloak authentication",
+  "dotnet-auth-jwt": "Use JSON Web Token (JWT) authentication",
+};
 
 const CreateServiceAuth: React.FC<WizardStepProps> = ({ formik }) => {
-  const AuthCoreSvg = ImgSvg({ image: authModuleImage });
+  const { pluginCatalog } = usePlugins(null, null);
+
+  const pluginList = useMemo(() => {
+    if (!pluginCatalog) {
+      return [];
+    }
+
+    if (formik.values.codeGenerator === EnumCodeGenerator.DotNet) {
+      return DOTNET_AUTH_PLUGINS.map((plugin) => pluginCatalog[plugin]).filter(
+        (plugin) => plugin
+      );
+    } else {
+      return NODE_AUTH_PLUGINS.map((plugin) => pluginCatalog[plugin]).filter(
+        (plugin) => plugin
+      );
+    }
+  }, [formik, pluginCatalog]);
+
+  useEffect(() => {
+    if (
+      pluginList &&
+      pluginList.length > 0 &&
+      !pluginList.find(
+        (plugin) => plugin.pluginId === formik.values.databaseType
+      )
+    ) {
+      formik.setFieldValue("databaseType", pluginList[0].pluginId);
```

Comment:
It feels a bit "patchy" to assume the DB plugin is first and count on it.

Reply:
and in general it's weird that "CreateServiceAuth" needs to know/care about DB plugins

Reply:
The idea is to set the first available plugin as the default if the current default is not available in the plugin list.
I used the wrong field name - I will fix it